torchci/pages/api/clickhouse/[queryName].ts

```diff
@@ -6,9 +6,25 @@ export default async function handler(
   res: NextApiResponse
 ) {
   const queryName = req.query.queryName as string;
-  const response = await queryClickhouseSaved(
-    queryName,
-    JSON.parse(req.query.parameters as string)
-  );
+
+  // detect if parameters are in the legacy Rockset lambda format
+  // (if they all contain "name", "type", and "value" keys)
+  // and convert them to the new format if so
+  let parameters = JSON.parse(req.query.parameters as string);
+
+  if (
```

Comment:
I guess this technique will works for string-alike type (string, int, datetime, etc) but couldn't handle complex types like Array or Tuple, i.e. https://github.com/pytorch/test-infra/pull/5576/files#diff-118f9a043b884ab6043546acc88078cf1e1298cf335f8434559cbfcb8f81bb7dR495.  Nevertheless, this works transparently, so it is still a nice tweak to have.

Reply:
Are you concerned about `req.query.parameters as string` or about specific non-string parameter values passed down to `queryClickhouseSaved`?